pkg/snapstore/gcs_snapstore.go

```diff
@@ -167,6 +167,18 @@ func (s *GCSSnapStore) Save(snap brtypes.Snapshot, rc io.ReadCloser) error {
 		return fmt.Errorf("failed uploading composite object for snapshot with error: %v", err)
 	}
 	logrus.Info("Composite object uploaded successfully.")
+
+	// Delete the chunks right after uploading the snapshot to bucket.
+	// To be done once the composite object is successfully uploaded.
+	logrus.Infof("Started deleting the chunk objects from bucket")
+	for partNumber := int64(1); partNumber <= noOfChunks; partNumber++ {
+		name := path.Join(prefix, snap.SnapDir, snap.SnapName, fmt.Sprintf("%010d", partNumber))
+		obj := bh.Object(name)
+		err := obj.Delete(context.TODO())
+		if err != nil {
+			return fmt.Errorf("failed to delete chunk with id: %d ", partNumber)
+		}
+	}
```

Comment:
- Encapsulate the chunk deletion logic into a dedicated function to improve code clarity, readability, and reusability.
- Utilize a context with a timeout when deleting chunks to prevent operations from hanging indefinitely.
- Accumulate errors encountered during chunk deletions instead of returning immediately upon the first error.
- Return all accumulated errors together after all deletion attempts, ensuring thorough cleanup and detailed feedback on the operation’s results.